source/common/umf_pools/disjoint_pool.cpp

```diff
@@ -293,6 +293,16 @@ class DisjointPool::AllocImpl {
         // Generate buckets sized such as: 64, 96, 128, 192, ..., CutOff.
         // Powers of 2 and the value halfway between the powers of 2.
         auto Size1 = params.MinBucketSize;
+        // Special case for when MinBucketSize equals 1 because otherwise bucket sizes starting from 1 value
+        // would generate with duplicates (like 1,1,2,2,4,4...) which is undesirable.
+        // In this case buckets are generated sized such as: 1, 2, 4, 8, 16, 32, 64, 96, 128 ..., Cutoff.
+        // Powers of 2 to the size of 64 and then powers of 2 and the values halfway
+        // between the powers of 2.
+        if (Size1 == 1) {
+            for (; Size1 < 64; Size1 *= 2) {
+                Buckets.push_back(std::make_unique<Bucket>(Size1, *this));
+            }
+        }
         auto Size2 = Size1 + Size1 / 2;
         for (; Size2 < CutOff; Size1 *= 2, Size2 *= 2) {
             Buckets.push_back(std::make_unique<Bucket>(Size1, *this));
```

Comment:
I don't like the way this is done. We are special-casing the default path (1) to calculate the bucket sizes differently than for other values. Wouldn't it be simpler to either do `auto Size1 = max(Size1, 2);` or, if we really want to have that bucket with 1 byte objects, something like `if Size1 == 1 { buckets.push_back(...); Size1 = 2; }`?

Reply:
Yeah, I think I like this better as well.

Reply:
Done.